src/ds/d500/d500-device.cpp

```diff
@@ -0,0 +1,838 @@
+
+        std::string device_name = (rs500_sku_names.end() != rs500_sku_names.find(_pid)) ? rs500_sku_names.at(_pid) : "RS5xx";
+
+        std::vector<uint8_t> gvd_buff(HW_MONITOR_BUFFER_SIZE);
+
+        auto& depth_sensor = get_depth_sensor();
+        auto& raw_depth_sensor = get_raw_depth_sensor();
+
+        using namespace platform;
+
+        std::string asic_serial, pid_hex_str, usb_type_str;
+        d500_gvd_parsed_fields gvd_parsed_fields;
+        bool advanced_mode = false;
+        bool usb_modality = true;
+        group_multiple_fw_calls(depth_sensor, [&]() {
+
+            _hw_monitor->get_gvd(gvd_buff.size(), gvd_buff.data(), ds::fw_cmd::GVD);
+
+            constexpr auto gvd_header_size = 8;
+            get_gvd_details(gvd_buff, &gvd_parsed_fields);
+            auto gvd_payload_data = gvd_buff.data() + gvd_header_size;
+            auto computed_crc = calc_crc32(gvd_payload_data, gvd_parsed_fields.payload_size);
+            LOG_INFO("gvd version = " << gvd_parsed_fields.gvd_version);
+            LOG_INFO("gvd payload size = " << gvd_parsed_fields.payload_size);
+            LOG_INFO("gvd crc = " << gvd_parsed_fields.crc32);
+            LOG_INFO("gvd optical module sn = " << gvd_parsed_fields.optical_module_sn);
+            if (computed_crc != gvd_parsed_fields.crc32)
+                LOG_ERROR("CRC mismatch in D500 GVD - received CRC = " << gvd_parsed_fields.crc32 << ", computed CRC = " << computed_crc);
+
+            _device_capabilities = ds_caps::CAP_ACTIVE_PROJECTOR | ds_caps::CAP_RGB_SENSOR | ds_caps::CAP_IMU_SENSOR |
+                ds_caps::CAP_BMI_085 | ds_caps::CAP_GLOBAL_SHUTTER | ds_caps::CAP_INTERCAM_HW_SYNC;
+
+            _fw_version = firmware_version(gvd_parsed_fields.fw_version);
+
+            auto _usb_mode = usb3_type;
+            usb_type_str = usb_spec_names.at(_usb_mode);
+            _usb_mode = raw_depth_sensor.get_usb_specification();
+            if (usb_spec_names.count(_usb_mode) && (usb_undefined != _usb_mode))
+                usb_type_str = usb_spec_names.at(_usb_mode);
+            else  // Backend fails to provide USB descriptor  - occurs with RS3 build. Requires further work
+                usb_modality = false;
+
+            _is_symmetrization_enabled = check_symmetrization_enabled();
+
+            depth_sensor.register_processing_block(processing_block_factory::create_id_pbf(RS2_FORMAT_Z16H, RS2_STREAM_DEPTH));
+
+            depth_sensor.register_processing_block(
+                { {RS2_FORMAT_Y8I} },
+                { {RS2_FORMAT_Y8, RS2_STREAM_INFRARED, 1} , {RS2_FORMAT_Y8, RS2_STREAM_INFRARED, 2} },
+                []() { return std::make_shared<y8i_to_y8y8>(); }
+            ); // L+R
+
+            depth_sensor.register_processing_block(
+                { RS2_FORMAT_Y16I },
+                { {RS2_FORMAT_Y16, RS2_STREAM_INFRARED, 1}, {RS2_FORMAT_Y16, RS2_STREAM_INFRARED, 2} },
+                []() {return std::make_shared<y16i_to_y10msby10msb>(); }
```

Comment:
@maloel please confirm Y16I format for this project

Reply:
We don't need it, but it's OK if it's in the dev branch

Reply:
I don't think so.

D400 --> User select Y16 --> we request Y12i from the camera
D500 --> User select Y16 --> we request Y16i from the camera

If this product use Y12I as calibration format you need to change this line.

Reply:
@maloel waiting for your answer here - if it is ok for you, please write

Reply:
Noted. For now I think it's fine.